backend/service/k8s/deployment.go

```diff
@@ -30,8 +30,11 @@ func (s *svc) DescribeDeployment(ctx context.Context, clientset, cluster, namesp
 }
 
 func ProtoForDeployment(cluster string, deployment *appsv1.Deployment) *k8sapiv1.Deployment {
+	if deployment.ClusterName == "" {
+		deployment.ClusterName = cluster
```

Comment:
this will modify the incoming object, which may not be desirable in some cases.

i think we should stick with the local var, override it with deployment.ClusterName if deployment.ClusterName not empty